spec/posts_index_spec.rb

```diff
@@ -0,0 +1,61 @@
+require 'rails_helper'
+
+RSpec.describe 'Post Index', type: :system do
+  before do
+    driven_by(:rack_test)
+  end
+
+  let(:user) do
+    User.create(name: 'Naan', photo: 'https://unsplash.com/photos/ABCDEF', posts_counter: 5)
+  end
+
+  let!(:post) do
+    Post.create(title: 'My First Post', text: 'This is my first post', author: user, comments_counter: 0,
+                likes_counter: 0)
+  end
+
+  let!(:comment) do
+    Comment.create(text: 'This is my first comment', author: user, post:)
+  end
+
+  it "displays the user's name on user posts page" do
+    visit users_path(user_id: user.id)
+    expect(page).to have_content(user.name)
+  end
+
+  it "displays the user's post count on user posts page" do
+    visit users_path(user_id: user.id)
+    expect(page).to have_content(user.posts_counter)
+  end
+
+  it 'displays a post title on user posts page' do
+    visit user_posts_path(user_id: user.id)
+    expect(page).to have_content(post.title)
+  end
+
+  it 'displays a post text on user posts page' do
+    visit user_posts_path(user_id: user.id)
+    expect(page).to have_content(post.text)
+  end
+
+  it 'displays the first comments on a post' do
+    visit user_posts_path(user_id: user.id)
+    expect(page).to have_content(comment.text)
+  end
+
+  it 'displays the number of comments on a post' do
+    visit user_posts_path(user_id: user.id)
+    expect(page).to have_content(post.comments_counter)
+  end
+
+  it 'displays the number of likes on a post' do
+    visit user_posts_path(user_id: user.id)
+    expect(page).to have_content(post.likes_counter)
+  end
+
+  it 'redirects to post show page when clicking on post title' do
+    visit user_posts_path(user_id: user.id)
+    click_link post.title
+    expect(page).to have_current_path(user_post_path(user_id: user.id, id: post.id))
+  end
```

Comment:
- [ ] @Team great job adding tests for the posting page, however, you forgot to add a test for the `that asserts the user can see the user's profile picture` and test `that asserts a user can see a section for pagination if there are more posts than fit on the view.

Reply:
- [ ] @naanmohammed and @amentetteh great work implementing all changes requested, however, forgot to do one requested change adding a test  `that asserts a user can see a section for pagination if there are more posts than fit on the view.`Kindly consider adding it.
![image](https://github.com/naanmohammed/Fintech-Forum/assets/61408860/a23e1bb8-caaf-45f6-93a8-fa14183b20b9)